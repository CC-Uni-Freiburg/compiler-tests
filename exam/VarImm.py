a = 96
b = -96
c = 19
d = -19
e = 0
print(a % 19) # 1
print(96 % c) # 1
print(0 % c) # 0
print(e % 19) # 0
print(b % 19) # -1
print(-96 % c) # -1
print(a % -19) # 1
print(96 % d) # 1
print(b % -19) # -1
print(-96 % d) # -1
print(0 % d) # 0
print(e % -19) # 0
# Jetz nochmal alles für maximal große Werte.
a = 9223372036854775807
b = -9223372036854775807
print(a % -9223372036854775807) # 0
print(9223372036854775807 % b) # 0
print(b % 9223372036854775807) # 0
print(-9223372036854775807 % a) # 0
print(0 % b) # 0
print(0 % a) # 0
