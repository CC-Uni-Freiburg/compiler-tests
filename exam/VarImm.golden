1100-1-111-1-100000000
